src/NBitcoin/ConcurrentChain.cs

```diff
@@ -71,7 +71,7 @@ public void Load(BitcoinStream stream)
                             this.blocksByHeight.Clear();
                             this.blocksById.Clear();
                             this.tip = null;
-                            this.SetTipLocked(new ChainedBlock(header, 0));
+                            this.SetTipLocked(new ChainedBlock(header, header.GetHash(), 0));
```

Comment:
Maybe we should pass options to this constructor

Reply:
This should be fine for now. Let me know if we can do that later.